### PR TITLE
Generic VM console test tooling

### PR DESCRIPTION
**Pull request by davidvossel**
I've stripped the work @rmohr did for his goexpect console tester in #499 and converted our existing tests to use it. 

I need this console tooling for some features I'm working on now. We shouldn't have to wait for node networking to land before taking advantage of this tooling improvement.

### CODE REVIEW

File: tests/console_test.go [hunk 1]
```diff
@@ -41,57 +37,28 @@ var _ = Describe("Console", func() {
...
-			Expect(err).ToNot(HaveOccurred())
-			return c
-		}
 	})
 
 	Context("New VM with a serial console given", func() {
+		It("should be returned that we are running cirros", func() {
+			vm := tests.NewRandomVMWithSerialConsole()
 
-		It("should be allowed to connect to the console", func(done Done) {
```
Comment (davidvossel) [inline tests/console_test.go]: Notice I removed this functional test entirely. I don't see any value in keeping a test that just verifies console access works when all our other console based tests exercise this.

If console access is broken, we'll be able to tell that by the location of the failure in our other console tests. We can tell if it failed at connecting or failed sending/receiving.

**Comment by davidvossel** [comment]
retest this please

**Comment by rmohr** [comment]
I think that #538 is not properly cleaned up and the port exposure does not work right now because of that.

**Comment by rmohr** [comment]
retest this please

**Pull request merged**